File: LeetCode/python/217_contains_duplicate.py
from typing import List
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        # Create a set and store each value
        # Loop through the list 
        # Check if there is a duplicate in the set 
        # If there is a duplicate, then break out of the loop and return true
        # Else return false 
        seen = set()
        for x in nums:
            if x in seen:
                return True
            seen.add(x)
        return False

File: LeetCode/typescript/src/217_contains_duplicate.ts
/**
 * Determines if the array contains any duplicate values.
 * @param nums - An array of integers
 * @returns true if any value appears at least twice, false otherwise
 */
function containsDuplicate(nums: number[]): boolean {
    const seen = new Set<number>();

    for (const num of nums) {
        if (seen.has(num)) {
            return true;
        }
        seen.add(num);
    }

    return false;
}